README.adoc

```diff
@@ -1,25 +1,74 @@
+. Click *Dependencies* and select *Task*, *H2 Database*, *JDBC API*, and *MariaDB Driver*.
+. Click *Generate*.
+. Download the resulting ZIP file, which is an archive of a task application that is configured with your choices.
+
+NOTE: If your IDE has the Spring Initializr integration, you can complete this process from your IDE.
+
+NOTE: You can also fork the project from Github and open it in your IDE or other editor.
+
 
 === What is Spring Cloud Task?
 A framework for building short-lived Spring Boot microservices, such as batch data processing jobs. You can learn more about
 the framework from the link:https://spring.io/projects/spring-cloud-task[project-site],
 link:https://spring.io/projects/spring-cloud-task#learn[documentation],
 and link:https://github.com/spring-cloud/spring-cloud-task/tree/main/spring-cloud-task-samples[samples].
 
-=== Simple Task Application
-In the following guide, we develop a Spring Boot application that uses Spring Cloud Task and deploy it to Cloud Foundry,
-Kubernetes, and your local machine.
+=== Task Application Overview
+In the following guide, we develop a Spring Boot application that uses Spring Cloud Task and deploy it to your local machine.
+We will use an H2 database to initially demonstrate Spring Cloud Task functionality, but the value of Spring Cloud Task
+is in its ability to store the results of executed tasks in a persistent datastore.  For this example we will progress by using MariaDB
+as our persistent datastore.
+
+=== Spring Cloud Task using In Memory H2
+For the first step, we will use Spring Cloud Task with an embedded H2 datastore.  If no `spring.datasource` properties are detected on project
+initialization, Spring Cloud Task will spin up a new H2 instance to store the executed task results.  This happens without any extra configuration requried.
+This facilitates a
+quick way to get up and running, but the results of your executed task will be lost when the application completes and
+the H2 database shuts down.  For this step in the guide, ensure that the `spring.datasource` properties in `application.properites` are commented out.
+
+=== Spring Cloud Task using Persistent MariaDB
+
+To persist data that lives longer than our short-lived task, we need an externally available datastore.  In this guide we will use a MariaDB instance running in a Docker container.  A version of Docker is required to be running locally in order to follow along.
+
+To run MariaDB locally, execute the following command:
+```
+docker run -p 3306:3306 --name mariadb -e MYSQL_ROOT_PASSWORD=password -e MYSQL_DATABASE=task -d mariadb:latest
+```
+
+When the MariaDB instance is running, supply the configuration to your application through `application.properties`.
 
-link:https://dataflow.spring.io/docs/batch-developer-guides/batch/spring-task/[Simple Task Application]
+```
+spring.datasource.url=jdbc:mariadb://localhost:3306/task
```

Comment:
Ideally, this would be an include from a file in the source tree.

Reply:
The lines referenced are documented out in the checked in code so I assume pulling them in as a reference would show them commented out.
The lines are commented out so that user can initially run the application without Docker running locally.

Reply:
We can live with it.